### PR TITLE
Type args bug fixes

### PR DESCRIPTION
**Pull request by rafaelbey**
#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

File: legend-engine-core/legend-engine-core-base/legend-engine-core-language-pure/legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/context/PackageableElementPointer.java [hunk 1]
```diff
@@ -81,15 +80,6 @@ public int hashCode()
     {
         return Objects.hash(type, path);
     }
-
```
Comment (rafaelbey) [inline legend-engine-core/legend-engine-core-base/legend-engine-core-language-pure/legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/context/PackageableElementPointer.java:84]: Check if studio can handle it.

File: legend-engine-core/legend-engine-core-base/legend-engine-core-language-pure/legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/domain/ProfileStereotype.java [hunk 2]
```diff
@@ -14,32 +14,26 @@
 
 package org.finos.legend.engine.protocol.pure.v1.model.packageableElement.domain;
 
-import com.fasterxml.jackson.databind.util.StdConverter;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
 import org.finos.legend.engine.protocol.pure.v1.model.SourceInformation;
 
 public class ProfileStereotype
```
Comment (rafaelbey) [inline legend-engine-core/legend-engine-core-base/legend-engine-core-language-pure/legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/domain/ProfileStereotype.java:21]: Check if studio can handle it.

File: legend-engine-core/legend-engine-core-base/legend-engine-core-language-pure/legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/domain/ProfileTag.java [hunk 3]
```diff
@@ -14,32 +14,26 @@
 
 package org.finos.legend.engine.protocol.pure.v1.model.packageableElement.domain;
 
-import com.fasterxml.jackson.databind.util.StdConverter;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
 import org.finos.legend.engine.protocol.pure.v1.model.SourceInformation;
 
 public class ProfileTag
```
Comment (rafaelbey) [inline legend-engine-core/legend-engine-core-base/legend-engine-core-language-pure/legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/domain/ProfileTag.java:21]: Check if studio can handle it.

File: legend-engine-core/legend-engine-core-base/legend-engine-core-language-pure/legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/mapping/mappingTest/StoreTestData.java [hunk 4]
```diff
@@ -15,16 +15,13 @@
...
 import org.finos.legend.engine.protocol.pure.v1.model.SourceInformation;
-import org.finos.legend.engine.protocol.pure.v1.model.context.PackageableElementPointer;
 import org.finos.legend.engine.protocol.pure.v1.model.data.EmbeddedData;
 import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.store.StoreProviderPointer;
 
 public class StoreTestData
 {
     public String doc;
-    @JsonSerialize(converter = PackageableElementPointer.ToPathSerializerConverter.class)
     public StoreProviderPointer store;
```
Comment (rafaelbey) [inline legend-engine-core/legend-engine-core-base/legend-engine-core-language-pure/legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/mapping/mappingTest/StoreTestData.java:25]: Check if studio can handle it.

File: legend-engine-core/legend-engine-core-base/legend-engine-core-language-pure/legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/domain/Profile.java [hunk 5]
```diff
@@ -14,18 +14,14 @@
...
 import java.util.List;
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.PackageableElement;
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.PackageableElementVisitor;
 
 public class Profile extends PackageableElement
 {
-    @JsonSerialize(contentConverter = ProfileStereotype.ToPathSerializerConverter.class)
     public List<ProfileStereotype> stereotypes = Collections.emptyList();
-    @JsonSerialize(contentConverter = ProfileTag.ToPathSerializerConverter.class)
     public List<ProfileTag> tags = Collections.emptyList();
```
Comment (rafaelbey) [inline legend-engine-core/legend-engine-core-base/legend-engine-core-language-pure/legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/domain/Profile.java:25]: Check if studio can handle it.

File: legend-engine-core/legend-engine-core-base/legend-engine-core-language-pure/legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/mapping/AssociationMapping.java [hunk 6]
```diff
@@ -16,22 +16,19 @@
...
-
 @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "_type")
 @JsonSubTypes({
         @JsonSubTypes.Type(value = XStoreAssociationMapping.class, name = "xStore")
 })
 public abstract class AssociationMapping
 {
     public String id;
-    @JsonSerialize(converter = PackageableElementPointer.ToPathSerializerConverter.class)
     public PackageableElementPointer association;
```
Comment (rafaelbey) [inline legend-engine-core/legend-engine-core-base/legend-engine-core-language-pure/legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/mapping/AssociationMapping.java:32]: Check if studio can handle it.

File: legend-engine-core/legend-engine-core-base/legend-engine-core-language-pure/legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/mapping/EnumerationMapping.java [hunk 7]
```diff
@@ -15,16 +15,13 @@
...
 import org.finos.legend.engine.protocol.pure.v1.model.SourceInformation;
 import org.finos.legend.engine.protocol.pure.v1.model.context.PackageableElementPointer;
 
-import java.util.List;
-
 public class EnumerationMapping
 {
     public String id;
-    @JsonSerialize(converter = PackageableElementPointer.ToPathSerializerConverter.class)
     public PackageableElementPointer enumeration;
```
Comment (rafaelbey) [inline legend-engine-core/legend-engine-core-base/legend-engine-core-language-pure/legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/mapping/EnumerationMapping.java:25]: Check if studio can handle it.

File: legend-engine-core/legend-engine-core-base/legend-engine-core-language-pure/legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/store/Store.java [hunk 8]
```diff
@@ -14,15 +14,12 @@
...
-
 import java.util.Collections;
 import java.util.List;
+import org.finos.legend.engine.protocol.pure.v1.model.context.PackageableElementPointer;
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.PackageableElement;
 
 public abstract class Store extends PackageableElement
 {
-    @JsonSerialize(contentConverter = PackageableElementPointer.ToPathSerializerConverter.class)
     public List<PackageableElementPointer> includedStores = Collections.emptyList();
```
Comment (rafaelbey) [inline legend-engine-core/legend-engine-core-base/legend-engine-core-language-pure/legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/store/Store.java:24]: Check if studio can handle it.

File: legend-engine-xts-persistence/legend-engine-xt-persistence-protocol/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/persistence/PersistenceContext.java [hunk 9]
```diff
@@ -14,19 +14,16 @@
...
 import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.PackageableElementVisitor;
 import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.connection.Connection;
 import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.persistence.context.PersistencePlatform;
 import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.persistence.service.parameter.ServiceParameter;
 
-import java.util.List;
-
 public class PersistenceContext extends PackageableElement
 {
-    @JsonSerialize(converter = PackageableElementPointer.ToPathSerializerConverter.class)
```
Comment (rafaelbey) [inline legend-engine-xts-persistence/legend-engine-xt-persistence-protocol/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/persistence/PersistenceContext.java:29]: Check if studio can handle it.

File: legend-engine-xts-persistence/legend-engine-xt-persistence-protocol/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/persistence/persister/sink/ObjectStorageSink.java [hunk 10]
```diff
@@ -14,12 +14,10 @@
 
 package org.finos.legend.engine.protocol.pure.v1.model.packageableElement.persistence.persister.sink;
 
-import com.fasterxml.jackson.databind.annotation.JsonSerialize;
 import org.finos.legend.engine.protocol.pure.v1.model.context.PackageableElementPointer;
 
 public class ObjectStorageSink extends Sink
 {
-    @JsonSerialize(converter = PackageableElementPointer.ToPathSerializerConverter.class)
```
Comment (rafaelbey) [inline legend-engine-xts-persistence/legend-engine-xt-persistence-protocol/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/persistence/persister/sink/ObjectStorageSink.java:22]: Check if studio can handle it.

File: legend-engine-xts-persistence/legend-engine-xt-persistence-protocol/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/persistence/persister/sink/RelationalSink.java [hunk 11]
```diff
@@ -14,12 +14,10 @@
 
 package org.finos.legend.engine.protocol.pure.v1.model.packageableElement.persistence.persister.sink;
 
-import com.fasterxml.jackson.databind.annotation.JsonSerialize;
 import org.finos.legend.engine.protocol.pure.v1.model.context.PackageableElementPointer;
 
 public class RelationalSink extends Sink
 {
-    @JsonSerialize(converter = PackageableElementPointer.ToPathSerializerConverter.class)
```
Comment (rafaelbey) [inline legend-engine-xts-persistence/legend-engine-xt-persistence-protocol/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/persistence/persister/sink/RelationalSink.java:22]: Check if studio can handle it.

File: legend-engine-xts-persistence/legend-engine-xt-persistence-target-relational-protocol/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/persistence/relational/sink/RelationalPersistenceTarget.java [hunk 12]
```diff
@@ -23,7 +22,6 @@
 public class RelationalPersistenceTarget extends PersistenceTarget
 {
     public String table;
-    @JsonSerialize(converter = PackageableElementPointer.ToPathSerializerConverter.class)
```
Comment (rafaelbey) [inline legend-engine-xts-persistence/legend-engine-xt-persistence-target-relational-protocol/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/persistence/relational/sink/RelationalPersistenceTarget.java:26]: Check if studio can handle it.

**Review by MauricioUyaguari — DISMISSED** [review]

**Review by MauricioUyaguari — APPROVED** [review]

File: legend-engine-xts-persistence/legend-engine-xt-persistence-protocol/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/persistence/PersistenceContext.java (hunk 9 above)
Comment (MauricioUyaguari) [reply legend-engine-xts-persistence/legend-engine-xt-persistence-protocol/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/persistence/PersistenceContext.java:29]: needs studio test - https://github.com/finos/legend-studio/pull/3713/

**Pull request merged**